src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -418,34 +424,66 @@ def app_events(
     last: int = typer.Option(
         default=0, help="Fetch only the last N events. Cannot be used with --first."
     ),
+    follow: bool = typer.Option(
+        False,
+        "--follow",
+        "-f",
+        help=f"Continue polling for events. Implies --last {DEFAULT_EVENT_FOLLOW_LAST} unless overridden.",
+    ),
+    follow_interval: int = typer.Option(
+        10,
+        help=f"Polling interval in seconds when using the --follow flag.",
+    ),
     **options,
 ):
     """Fetches events for this app from the event table configured in Snowflake."""
     if first and last:
         raise ClickException("--first and --last cannot be used together.")
 
+    if follow:
+        if until:
+            raise ClickException("--follow and --until cannot be used together.")
+        if first:
+            raise ClickException("--follow and --first cannot be used together.")
+
     assert_project_type("native_app")
 
+    record_type_names = [r.name for r in record_types]
     manager = NativeAppManager(
         project_definition=get_cli_context().project_definition.native_app,
         project_root=get_cli_context().project_root,
     )
-    events = manager.get_events(
-        since_interval=since,
-        until_interval=until,
-        record_types=[r.name for r in record_types],
-        scopes=scopes,
-        first=first,
-        last=last,
-    )
-    if not events:
-        return MessageResult("No events found.")
-
-    def g():
-        for event in events:
-            yield EventResult(event)
+    if follow:
+        if not last:
+            # If we don't have a value for --last, assume 20 so we
+            # at least print something before starting the tail
+            last = DEFAULT_EVENT_FOLLOW_LAST
```

Comment:
Not sure about this, in the sense that if I use `snow app events --since "10 mins" --follow`, I'd expect all events matching my query, then a follow. Sounds like we're preventing that here?

I'm ok with some small number if the user specifies nothing else though, that makes sense. We might want to do this anyway if the user just says `snow app events` without --follow, for consistency.

Reply:
good point, i'll rework the UX of the flags when `--follow` is used